extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js

```diff
@@ -359,4 +394,25 @@ describes.realWin('amp-ad-exit', {
         .to.have.been.calledWith(
         'http://localhost:8000/tracking?bar=bar', '');
   });
+
+  it('should replace custom URL variables with 3P Analytics signals', () => {
+    const open = sandbox.stub(win, 'open', () => {
+      return {name: 'fakeWin'};
+    });
+
+    addToResponseMap(env.ampdoc, TEST_3P_VENDOR, env.win.document.baseURI, {
+      'unused': 'unused',
+      'collected-data': 'abc123',
+    });
+
+    element.implementation_.executeAction({
+      method: 'exit',
+      args: {target: 'variableFrom3pAnalytics'},
+      event: makeClickEvent(1001, 101, 102),
+      satisfiesTrust: () => true,
+    });
+
+    expect(open).to.have.been.calledWith(
+        'http://localhost:8000/vars?foo=abc123', '_blank');
+  });
```

Comment:
TODO, could just as easily be in a different PR: Add:
it('should fail if value of vendorAnalyticsSource not found in vendors.js')